app/controllers/map_controller.rb

```diff
@@ -42,4 +42,34 @@ def directions
     res = Net::HTTP.get_response(uri)
     render json: res.body
   end
+
+  def url
+    profile = params[:profile]
```

Comment:
The code from here to the end of the file is duplicated.

Reply:
fixed it in 4860ac08b0adf6acb7cde1ddef0a22d1b10063d5